### DISCUSSION TITLE
Does Budibase include a Kanban board component?

### DISCUSSION THREAD

**Discussion by mattfysh** [Budibase/budibase#6469] [General]
title

**Comment by melohagan** [comment]
Hi @mattfysh 

The short answer is no, however *Custom Components* are in Q3 of the roadmap: https://github.com/Budibase/budibase/projects/10

This is highly requested functionality that will allow Budibase developers to create a wide range of custom svelte components, which could definitely include a dynamic Kanban board.

With that said, currently it is straightforward to build a rudimentary Kanban board:

![kanban](https://user-images.githubusercontent.com/101575380/176927407-05079ddc-5021-42bc-a55d-3a9db732ab09.gif)
 
### Tickets
1. Create a table for your tickets and create the autogenerated screens for the tickets table. One of the columns should be *Options* type to hold the different statuses.
2. In the edit screen `/tickets/:id` edit the 'Save' button actions and add the **Close Screen Modal** action as the last action.

### Kanban
1. Add a new screen for the Kanban board (set URL to`/` for home screen)
2. Add a title and a button that links you to the new ticket screen
3. Add a *Row direction* container which will hold the three panels.

### Backlog Panel
1. Add a container to the Kanban container with width = 33%
2. Add a Stat Card with the title 'Backlog'
3. Underneath add a Data Provider for the Tickets table. Define filters: *Status Equals Backlog*
4. Add a Repeater to the Data Provider and nest a Card under that.
5. Set the *Title*, *Subtitle*, and *Description*. Set the link URL to `/tickets/{{ Repeater.Tickets._id }}`. Tick *Open link in modal* and also tick *Show button* with text `⇒`
6. Define button actions for the Card: **Save Row** and 'Add Column' - *Status = In Progress*

### In Progress & Done panels
1. Duplicate the panel container above twice (once for In Progress and another for Done)
2. Remove the button from the Card in the Done panel.
3. Set the background colours as desired.

That's it! Here's the app export if you want to take a look: 
[Kanban-export-1656691137310.txt](https://github.com/Budibase/budibase/files/9030148/Kanban-export-1656691137310.txt)

**Reply by tw02d008** [reply]
Thanks for the great sample.
Saved me a lot of time and works flawless 👍

**Reply by melohagan** [reply]
Glad you found it helpful! Since this post we've added in [custom components](https://docs.budibase.com/docs/custom-component).
This would allow for a much smoother Kanban experience, and allow for more fine grain customization. 

With that said, I'm not aware of a Kanban component that has currently been built, but it certainly is possible.
